tsinfer/inference.py

```diff
@@ -1174,7 +1174,14 @@ def __ancestor_find_path(self, ancestor, thread_index=0):
         end = ancestor.end
         assert ancestor.haplotype.shape[0] == (end - start)
         haplotype[start:end] = ancestor.haplotype
-        self._find_path(ancestor.id, haplotype, start, end, thread_index)
+        try:
+            self._find_path(ancestor.id, haplotype, start, end, thread_index)
+        except _tsinfer.LibraryError as e:
+            if "Match impossible" in str(e):
```

Comment:
Matching on the strerror is brittle. What we should do is add specific exception in the C API and raise this exception when it happens. We might even be able to do this at the Python-C API directly, and avoid some boilerplate.

Actually, that's what we should do.

Reply:
OK - I'll leave it with you. Thanks